vkHW/vkHW/Views/Friends/LikeControl.swift

```diff
@@ -0,0 +1,112 @@
+// LikeControl.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+///  Контрол количества лайков
+class LikeControl: UIControl {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let defaultBlackColor = "defaultTextColor"
+        static let heartImageName = "heart"
+        static let heartFillImageName = "heart.fill"
+    }
+
+    // MARK: - Public Properties
+
+    var likeCount: Int = 0 {
+        didSet {
+            likeCountLabel.text = String(likeCount)
+        }
+    }
+
+    var isLiked: Bool = true {
+        didSet {
+            likeCountLabel.text = String(likeCount)
+            if isLiked {
+                heartImageView.image = UIImage(systemName: Constants.heartImageName)
+            } else {
+                heartImageView.image = UIImage(systemName: Constants.heartFillImageName)
+            }
+        }
+    }
+
+    // MARK: - Private Properties
+
+    private var heartImageView = UIImageView()
+    private var likeCountLabel = UILabel()
+
+    // MARK: - Initializers
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupUI()
+    }
+
+    // MARK: - Private Methods
+
+    @objc private func touchUpInside() {
+        if isLiked {
```

Comment:
```suggestion
        if isLiked {
```
используй терновник не дублируй код